source/geometry_model/initial_topography_model/prm_polygon.cc

```diff
@@ -0,0 +1,157 @@
+/*
+  Copyright (C) 2011 - 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/geometry_model/initial_topography_model/prm_polygon.h>
+#include <aspect/utilities.h>
+#include <boost/lexical_cast.hpp>
+
+namespace aspect
+{
+  namespace InitialTopographyModel
+  {
+    template <int dim>
+    double
+    PrmPolygon<dim>::
+    value (const Point<dim-1> &p) const
+    {
+      double value = 0;
+      Point<2> p1;
+
+      if (dim == 2)
+        p1 = Point<2>(p[0],0);
+      else
+        p1 = Point<2>(p[0],p[1]);
+
+      for (unsigned int i = 0; i < point_lists.size(); i++)
+        {
+          if (aspect::Utilities::polygon_contains_point<dim>(point_lists[i],p1))
+            {
+              value = topography_values[i];
+            }
+        }
```

Comment:
The previous comments on this code have been erased during renaming of the function, but to rehash: I said that this is inefficient because it has to go through all of the point lists, and then takes the value of the *last* polygon that matches. You said that this is the intended behavior.

I think I don't understand your use case where you have multiple polygons, and more than one of them actually has the given point inside. If that's the case, why take the *last* polygon's value, rather than the *first*? Taking the first has the advantage that you can just exit the function at this point, rather than having to search through all following polygons.